tsdb/index/index_test.go

```diff
@@ -719,3 +719,35 @@ func TestChunksTimeOrdering(t *testing.T) {
 
 	require.NoError(t, idx.Close())
 }
+
+func TestReader_PostingsForLabelMatchingHonorsContextCancel(t *testing.T) {
+	dir := t.TempDir()
+
+	idx, err := NewWriter(context.Background(), filepath.Join(dir, "index"))
+	require.NoError(t, err)
+
+	for i := 1; i <= 1000; i++ {
+		require.NoError(t, idx.AddSymbol(fmt.Sprintf("%4d", i)))
+	}
+	require.NoError(t, idx.AddSymbol("__name__"))
+
+	for i := 1; i <= 1000; i++ {
+		require.NoError(t, idx.AddSeries(storage.SeriesRef(i), labels.FromStrings("__name__", fmt.Sprintf("%4d", i)),
+			chunks.Meta{Ref: 1, MinTime: 0, MaxTime: 10},
+		))
+	}
+
+	require.NoError(t, idx.Close())
+
+	ir, err := NewFileReader(filepath.Join(dir, "index"))
+	require.NoError(t, err)
+	defer ir.Close()
+
+	failAfter := uint64(500)
```

Comment:
[super nit] I would do it like this, and then the same in other places. This way we enforce that the value used here is in sync with `checkContextEveryNIterations` otherwise if it's not divisibly by `checkContextEveryNIterations` then the assertion `require.Equal(t, failAfter, ctx.Count())` (below) will fail.

```suggestion
	failAfter := uint64(5 * checkContextEveryNIterations)
```

Reply:
yes, was about to do it, but meetings galore